vip-cache-manager/vip-cache-manager.php

```diff
@@ -84,8 +86,40 @@ public function manual_purge_message() {
 	public function curl_multi( $requests ) {
 		$curl_multi = curl_multi_init();
 
-		foreach ( $requests as $req ) {
-			if ( defined( 'PURGE_SERVER_TYPE' ) && 'mangle' == PURGE_SERVER_TYPE ) {
+		if ( defined( 'PURGE_BATCH_SERVER_URL' ) && defined( 'PURGE_SERVER_TYPE' ) && 'mangle' == PURGE_SERVER_TYPE ) {
+			$req_chunks = array_chunk( $requests, self::CACHE_PURGE_BATCH_SIZE, true );
+			foreach ( $req_chunks as $req_chunk ) {
+				$req_array = array();
+				foreach ( $req_chunk as $req ) {
+					$req_array[] = array(
+						'group' => 'vip-go',
+						'scope' => 'global',
+						'type'  => $req['method'],
+						'uri'   => $req['host'] . $req['uri'],
+					);
+				}
+				$data = json_encode( $req_array );
+
+				$curl = curl_init( constant( 'PURGE_BATCH_SERVER_URL' ) );
+
+				curl_setopt( $curl, CURLOPT_HEADER, false );
+				curl_setopt( $curl, CURLOPT_RETURNTRANSFER, true );
+				curl_setopt( $curl, CURLOPT_TIMEOUT, 5 );
+				curl_setopt( $curl, CURLOPT_POST, true );
+
+				if ( 500 < strlen( $data ) ) {
```

Comment:
Is the concern here the potential overhead of gzipping something? With this data structure, it only takes about 5-6 URLs to go over 500. Here's my unscientific experiment: https://repl.it/@rinatkhaziev/KnowledgeableRoastedParentheses#main.php

Maybe it's not worth having its own `if/else` block, and we could just gzip regardless of the size of the batch.

Reply:
Yes, the idea was avoiding overhead from gzip operations. Compressing a small amount of data may increase the size of what is sent on the wire, and it will also incur a latency penalty due to the gzip operations.